packages/components/field-label/src/field-label.tsx

```diff
@@ -83,6 +83,28 @@ export type TFieldLabelProps = {
     | 'auto';
 };
 
+/*
+  This is needed to deal with every FieldLabel row vertical spacing
+  when provided `hint` or `description` is a React component which
+  might render nothing.
+  Previously we were using the `Stack` component but, as we are wrapping
+  those props values with some elements, `Stack` was including some vertical
+  spacing even when the received values did not render anything.
+
+  The implementation is tightly coupled to how we currently wrap
+  those props, so if we change that, we will need to adjust this as well.
+*/
+const LabelRowWrapper = styled.div`
```

Comment:
This is the new component we use to conditionally add the vertical spacing between `FieldLabel` *rows*.

First selector is for the `hint` wrapper and the second one is for the `description` one.